drivers/scale_host.go

```diff
@@ -0,0 +1,383 @@
+		return http.StatusBadRequest, fmt.Errorf("No host with label %v exists", selectorLabel)
+	}
+
+	// Consider the least recently created as base host for cloning
+	// Remove domain from host name, scaleHost12.foo.com becomes scaleHost12
+	// Remove largest number suffix from end, scaleHost12 becomes scaleHost
+	// Name has precedence over hostname. If name is set, empty this field for the clones
+	host := hostScalingGroup[len(hostScalingGroup)-1]
+	if host["name"] != nil {
+		baseHostName = host["name"].(string)
+		host["name"] = nil
+	} else {
+		baseHostName = host["hostname"].(string)
+	}
+	baseHostName = strings.Split(baseHostName, ".")[0]
+	re := regexp.MustCompile("[0-9]+$")
+	baseSuffix := re.FindString(baseHostName)
+	basePrefix := strings.TrimRight(baseHostName, baseSuffix)
+
+	hostCreateURL := cattleURL + "/projects/" + host["accountId"].(string) + "/hosts"
+	count = 0
+
+	if action == "up" {
+		newHostScale = amount + int64(len(hostScalingGroup))
+		if newHostScale > max {
+			return http.StatusBadRequest, fmt.Errorf("Cannot scale above provided max scale value")
+		}
+
+		for _, currentHost := range hostScalingGroup {
+			if currentHost["name"] != nil {
+				currCloneName = currentHost["name"].(string)
+			} else {
+				currCloneName = currentHost["hostname"].(string)
+			}
+
+			if !strings.Contains(currCloneName, basePrefix) {
+				continue
+			}
+
+			currCloneName = strings.Split(currCloneName, ".")[0]
+			re := regexp.MustCompile("[0-9]+$")
+			suffix = re.FindString(currCloneName)
+
+			break
+		}
+
+		for count < amount {
+			if suffix != "" {
+				prevNumber, err := strconv.Atoi(suffix)
+				if err != nil {
+					return http.StatusInternalServerError, fmt.Errorf("Error converting %s to int : %v", suffix, err)
+				}
+				currNumber := prevNumber + 1
+				currNameSuffix = leftPad(strconv.Itoa(currNumber), "0", len(suffix))
+			} else {
+				currNameSuffix = "-2"
+			}
+
+			name := basePrefix + currNameSuffix
+			host["hostname"] = name
+			labels := make(map[string]interface{})
+			labels[key] = value
+			host["labels"] = labels
+
+			code, err := createHost(host, hostCreateURL, httpClient)
+			if err != nil {
+				return code, err
+			}
+
+			suffix = currNameSuffix
+			count++
+		}
+	} else if action == "down" {
+		newHostScale = int64(len(hostScalingGroup)) - amount
+		if newHostScale < min {
+			return http.StatusBadRequest, fmt.Errorf("Cannot scale above below min scale value")
+		}
+
+		if len(hostScalingGroup) == 1 {
+			return http.StatusBadRequest, fmt.Errorf("Cannot scale below 1")
+		}
+
+		deleteHosts := []map[string]interface{}{}
+		if deleteOption == "mostRecent" {
+			for count < amount {
+				deleteHosts = append(deleteHosts, hostScalingGroup[count])
+				count++
+			}
+		} else if deleteOption == "leastRecent" {
+			for count < amount {
+				index = (int64(len(hostScalingGroup)) - count) - 1
+				deleteHosts = append(deleteHosts, hostScalingGroup[index])
+				count++
+			}
+		}
+
```

Comment:
In the for loop below, for using API client, I would first have to get host ById. Since I'm already getting all hosts I thought I'll use http calls instead of getting hosts again and then deleting them.